aptos-move/framework/aptos-framework/sources/account/account.move

```diff
@@ -698,17 +746,32 @@ module aptos_framework::account {
 
     /// Revoke any signer capability offer in the specified account.
     public entry fun revoke_any_signer_capability(account: &signer) acquires Account {
+        let offerer_addr = signer::address_of(account);
+        if (features::is_default_account_resource_enabled()) {
+            assert!(
+                resource_exists_at(offerer_addr),
+                error::not_found(ENO_SUCH_SIGNER_CAPABILITY),
+            );
+        } else {
+            assert!(exists_at(offerer_addr), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+        };
         check_offering_permission(account);
-        let account_resource = borrow_global_mut<Account>(signer::address_of(account));
+        let account_resource = borrow_global_mut<Account>(offerer_addr);
         option::extract(&mut account_resource.signer_capability_offer.for);
     }
 
     /// Return an authorized signer of the offerer, if there's an existing signer capability offer for `account`
     /// at the offerer's address.
     public fun create_authorized_signer(account: &signer, offerer_address: address): signer acquires Account {
         check_offering_permission(account);
-        assert!(exists_at(offerer_address), error::not_found(EOFFERER_ADDRESS_DOES_NOT_EXIST));
-
+        if (features::is_default_account_resource_enabled()) {
+            assert!(
+                resource_exists_at(offerer_address),
+                error::not_found(ENO_SUCH_SIGNER_CAPABILITY),
+            );
+        } else {
+            assert!(exists_at(offerer_address), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+        };
```

Comment:
this seems like a common check , only difference is error type - maybe create a function that receives error type (ENO_SUCH_SIGNER_CAPABILITY) and does the check?